decidim-core/spec/helpers/decidim/action_authorization_helper_spec.rb

```diff
@@ -5,150 +5,147 @@
 module Decidim
   describe ActionAuthorizationHelper do
     let!(:component) { create(:component) }
-    let!(:user) { create(:user) }
-    let!(:action) { "foo" }
-
-    let(:status) do
-      double(
-        handler_name: handler_name,
-        code: code,
-        data: data,
-        "ok?": ok?
-      )
-    end
-
-    let(:ok?) { false }
-    let(:code) { :missing }
-    let(:handler_name) { "foo_authorization" }
-    let(:data) { { action: data_action, fields: data_fields } }
-    let(:data_action) { :authorize }
-    let(:data_fields) { nil }
+    let(:user) { create(:user) }
+    let(:action) { "foo" }
+    let(:authorized) { true }
 
     before do
+      allow(helper).to receive(:current_component).and_return(component)
       allow(helper).to receive(:current_user).and_return(user)
-      allow(helper).to receive(:action_authorization).with(action).and_return(status)
+      allow(helper).to receive(:action_authorized_to).with(action).and_return(authorized)
+      allow(helper).to receive(:action_authorized_to).with(nil).and_return(authorized)
     end
 
-    describe "action_authorization_modal" do
-      context "when missing" do
-        let(:code) { :missing }
-        let(:ok?) { false }
-
-        it "renders a modal with the missing information" do
-          expect(helper).to receive(:authorize_action_path).with("foo").and_return "authorization_route"
-          rendered = helper.action_authorization_modal("foo")
-          expect(rendered).to include("missing-authorization")
-          expect(rendered).not_to include("expired-authorization")
-          expect(rendered).not_to include("incomplete-authorization")
-          expect(rendered).not_to include("unauthorized-authorization")
+    describe "action_authorized_link_to" do
+      context "when the action is authorized" do
+        it "renders a regular link" do
+          rendered = helper.action_authorized_link_to("foo", "Link", "fake_path")
+          expect(rendered).not_to include("data-open")
+          expect(rendered).to include("<a")
+          expect(rendered).to include("Link")
+        end
+
+        it "renders with a block" do
+          rendered = helper.action_authorized_link_to("foo", "fake_path") { "Link" }
+
```

Comment:
Super tiny style note, you might want to remove this blank line for consistency with the other specs (although I think visually splitting non assertions from assertions reads better, so you might as well add a blank line to the other specs).

Reply:
I agree with you that it reads better with an extra space there, and I don't see a clear guideline in other spec files, so I'm adding blank lines to every other specs

Reply:
I like having non-assertions and assertions separate, but It's a rule I don't enforce even on myself so... 🤷‍♂️ There's some inconsistency throughout the code about this.

Reply:
Yeah, since `rubocop-rspec` don't support it, it's hard to enforce. Maybe I'll do a feature request :)